### DISCUSSION TITLE
datepicker filter value changes when page is refresh

### DISCUSSION THREAD

**Discussion by kjjd84** [filamentphp/filament#14822] [Help]
### Package

Panel builder

### Package Version

3.2

### How can we help you?

when i select a date via a datepicker inside my filter dropdown, and then refresh the page, the value of the datepicker / filter changes.

see this video:

https://github.com/user-attachments/assets/2b3c3e9d-0355-4ce6-b011-6e492cd8191f

**Comment by francoism90** [comment]
@kjjd84 Did you found a solution? I have the same issue, it seems to incorrectly convert the timezone.

**Reply by kjjd84** [reply]
one guy says casting the dates a certain way fixes the issue, but that was 2 years ago in filament v2:

https://github.com/filamentphp/filament/issues/5601#issuecomment-1680371753

not sure if it works.

**Reply by francoism90** [reply]
@kjjd84 Thanks! I'll test if it works, otherwise I will create a bug report. :)

**Reply by kjjd84** [reply]
> @kjjd84 Thanks! I'll test if it works, otherwise I will create a bug report. :)

@francoism90 please let me know if it works. i need to fix this on several crms myself.

**Reply by francoism90** [reply]
@kjjd84 Sorry forgot to reply.. unfortunately it didn't fix the problem for me. :/

**Reply by kjjd84** [reply]
@francoism90 

Using the native DatePicker fixes the issue. Just don't use ->native(false) on the DatePicker in your custom filter, and it works. The non-native datepicker appends the timezone offset which changes on refresh etc. The native datepicker does not.

**Comment by kjjd84** [comment]
@danharrin 

i figured out this bug. if you use the native datepicker this does not happen. the non-native datepicker seems to append the timezone offset to the filter value, which changes each time you refresh the page

so i am just using the native datepicker from now on

**Comment by diegosouzapw** [comment]
Hey @Alph4d0g, this is amazing — thank you for building this! 🎉

Great news: **OpenCode is already registered as a built-in agent in OmniRoute's ACP registry** (since v2.0.11). If you run OmniRoute and have `opencode` installed, it'll auto-detect it on the Agents page.

Your plugin is doing exactly the right thing with the `/connect` flow + `@ai-sdk/openai-compatible`. Here's how it maps to OmniRoute's native config:

```json
// opencode.json (auto-generated by OmniRoute in a future update)
{
  "$schema": "https://opencode.ai/config.json",
  "provider": {
    "omniroute": {
      "npm": "@ai-sdk/openai-compatible",
      "name": "OmniRoute",
      "options": {
        "baseURL": "http://localhost:4444/v1",
        "apiKey": "YOUR_OMNIROUTE_API_KEY"
      },
      "models": { /* fetched dynamically from /v1/models */ }
    }
  }
}
```

**What's coming natively:**
- The Agents dashboard will have a **"Generate opencode.json"** button when opencode is detected as installed
- It will auto-fill `baseURL`, `apiKey`, and fetch models from `/v1/models` — essentially the same thing your plugin does, but from within OmniRoute's UI

In the meantime your plugin is the perfect solution. I'll link it in the README under the OpenCode section. Thanks again! 🙏

**Comment by diegosouzapw** [comment]
Hey @Alph4d0g — just shipped it! 🎉

The **"Download opencode.json"** button is now live in the Agents dashboard. It:
- Appears automatically when `opencode` is detected as installed on your system
- Fetches all available models from `/v1/models` in real-time
- Generates a complete `opencode.json` config with the OmniRoute provider, your base URL, and the full model list
- Downloads the file directly to your machine — just drop it in your project root and run `opencode`

This will be included in the next release (v2.0.17). Your plugin remains the best option for the `/connect` flow, and we'll link to it from the docs as the recommended way to connect interactively.

Thanks again for building it and sharing it here — it directly inspired this native integration! 🙌

Track it in: https://github.com/diegosouzapw/OmniRoute/pull/254
